base/range_iterator.hpp

```diff
@@ -0,0 +1,101 @@
+#include "std/iterator.hpp"
+#include "std/type_traits.hpp"
+
+namespace base
+{
+
+// RagneItrator allows to write for loops as follows:
+// for (auto const i : range(N))
+//   ...
+// for (auto const i : range(M, N))
+//   ...
+// And initialize stl containers like this:
+// vector<int> mySequence(MakeRangeIterator(0), MakeRangeIterator(10));
+template <typename TCounter>
+struct RangeIterator
+{
+  explicit RangeIterator(TCounter const current) : m_current(current) { }
+
+  RangeIterator & operator++() { ++m_current; return *this; }
+  RangeIterator & operator--() { --m_current; return *this; }
```

Comment:
It's up to you. I think it's quite easy to implement them, so why not?

BTW, clang-format this file, please (and others, if needed). Implementations for operator++ and operator-- are not one-liners, so please, don't write them in one line.